website/docs/guides/eks-getting-started.html.md

```diff
@@ -0,0 +1,594 @@
+}
+
+resource "aws_iam_instance_profile" "demo-node" {
+  name = "terraform-eks-demo"
+  role = "${aws_iam_role.demo-node.name}"
+}
+```
+
+#### Worker Node Security Group
+
+This security group controls networking access to the Kubernetes worker nodes.
+
+```hcl
+resource "aws_security_group" "demo-node" {
+  name        = "terraform-eks-demo-node"
+  description = "Security group for all nodes in the cluster"
+  vpc_id      = "${aws_vpc.demo.id}"
+
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  tags = "${
+    map(
+     "Name", "terraform-eks-demo-node",
+     "kubernetes.io/cluster/${var.cluster-name}", "owned",
+    )
+  }"
+}
+
+resource "aws_security_group_rule" "demo-node-ingress-self" {
+  description              = "Allow node to communicate with each other"
+  from_port                = 0
+  protocol                 = "-1"
+  security_group_id        = "${aws_security_group.demo-node.id}"
+  source_security_group_id = "${aws_security_group.demo-node.id}"
+  to_port                  = 65535
+  type                     = "ingress"
+}
+
+resource "aws_security_group_rule" "demo-node-ingress-cluster" {
+  description              = "Allow worker Kubelets and pods to receive communication from the cluster control plane"
+  from_port                = 1025
+  protocol                 = "tcp"
+  security_group_id        = "${aws_security_group.demo-node.id}"
+  source_security_group_id = "${aws_security_group.demo-cluster.id}"
+  to_port                  = 65535
+  type                     = "ingress"
+}
+```
+
+#### Worker Node Access to EKS Master Cluster
+
+Now that we have a way to know where traffic from the worker nodes is coming
+from, we can allow the worker nodes networking access to the EKS master cluster.
+
+```hcl
+resource "aws_security_group_rule" "demo-cluster-ingress-node-https" {
+  description              = "Allow pods to communicate with the cluster API Server"
+  from_port                = 443
+  protocol                 = "tcp"
+  security_group_id        = "${aws_security_group.demo-cluster.id}"
+  source_security_group_id = "${aws_security_group.demo-node.id}"
+  to_port                  = 443
+  type                     = "ingress"
+}
+```
+
+#### Worker Node AutoScaling Group
+
+Now we have everything in place to create and manage EC2 instances that will
+serve as our worker nodes in the Kubernetes cluster. This setup utilizes
+an EC2 AutoScaling Group (ASG) rather than manually working with EC2 instances.
+This offers flexibility to scale up and down the worker nodes on demand when
+used in conjunction with AutoScaling policies (not implemented here).
+
+First, let us create a data source to fetch the latest Amazon Machine Image
+(AMI) that Amazon provides with an EKS compatible Kubernetes baked in.
+
+```hcl
+data "aws_ami" "eks-worker" {
+  filter {
+    name   = "name"
+    values = ["eks-worker-*"]
```

Comment:
Just curious - this is version agnostic intentionally? i.e. is the data source always going to fetch a valid AMI regardless of the K8S version (currently 1.10)?

Reply:
I think we can address this at a later time by updating the various pieces that might be version specific. 👍  Currently the guide is version agnostic.